torchvision/transforms/transforms.py

```diff
@@ -629,3 +629,61 @@ def __call__(self, img):
         angle = self.get_params(self.degrees)
 
         return F.rotate(img, angle, self.resample, self.expand, self.center)
+
+
+class Grayscale(object):
+    """Convert image to grayscale.
+    Args:
+        num_output_channels (int): (1 or 3) number of channels desired for output image
+
+    Returns:
+        PIL Image: grayscale version of the input
+                   if num_output_channels == 1 : returned image is single channel
+                   if num_output_channels == 3 : returned image is 3 channel with r == g == b
+
+    """
+
+    def __init__(self, num_output_channels=1):
+        self.num_output_channels = num_output_channels
+
+    def __call__(self, img):
+        """
+        Args:
+            img (PIL Image): Image to be converted to grayscale.
+
+        Returns:
+            PIL Image: Randomly grayscaled image.
+        """
+        return F.to_grayscale(img, num_output_channels=self.num_output_channels)
+
+
+class RandomGrayscale(object):
+    """Randomly convert image to grayscale with a probability of p (default 0.1).
+    Args:
+        p (float): probability that image should be converted to grayscale.
+
+    Returns:
+        PIL Image: grayscale version of the input image with probability p
+                   and unchanged with probability (1-p)
+                   - if input image is 1 channel:
+                        grayscale version is 1 channel
+                   - if input image is 3 channel:
+                        grayscale version is 3 channel with r == g == b
+
+    """
+
+    def __init__(self, p=0.1):
```

Comment:
I think `num_output_channels` should be passed to the constructor to be consistent with `Grayscale`

Reply:
For RandmGrayscale() what would transform of  p=0.1, num_output_channels=1 mean for a batch of RGB images ? For a large enough batch, on an average, 10% get converted to single channel grayscale and 90% remain unchanged i.e 3 channel. This would be problematic. Hence, the restriction of not changing number of output channels for RandomGrayscale()

On the other hand, for Grayscale() is a deterministic transform to convert to grayscale, where you can have 1-single channel output or 3-channel output.

Hence, the interfaces are not consistent.

Reply:
hmm, yeah that makes sense 👍